src/App.js

```diff
@@ -1,65 +1,44 @@
 import React, { Component, Fragment } from "react";
 import { Switch, Route } from "react-router-dom";
 
-import Header from "./components/Header/Header";
-import Homepage from "./pages/Homepage/Homepage";
-import Favspage from "./pages/Favspage/Favspage";
-import MovieInfo from "./pages/MovieInfo/MovieInfo";
+import Header from "./components/Header";
+import Homepage from "./pages/Homepage";
+import Favspage from "./pages/Favspage";
+import MovieInfo from "./pages/MovieInfo";
 
 import "./App.css";
 
 class App extends Component {
-  constructor() {
-    super();
-
-    this.state = {
-      movies: "",
-      query: "",
-      favs: localStorage.favs ? JSON.parse(localStorage.favs) : []
-    };
-  }
-
-  componentDidMount() {
-    console.log(this.state.favs)
-  }
-  
+  state = {
+    movies: "",
+    query: "",
+    favs: localStorage.favs ? JSON.parse(localStorage.favs) : []
+  };
 
   saveInLocalStorage = favs => {
     localStorage.setItem("favs", JSON.stringify(favs));
   };
 
-  handleFavButttonClicked = event => {
+  handleFavButtonClicked = event => {
     const value = event.currentTarget.value;
-    
-    if (!this.state.favs.includes(value)) {
-      this.setState(
-        state => {
-          const favs = state.favs.concat(value);
+    this.setState(
+      state => {
+        const favs = state.favs.includes(value)
+          ? state.favs.filter(id => id !== value)
+          : state.favs.concat(value);
 
-          return {
-            favs
-          };
-        },
-        () => {
-          this.saveInLocalStorage(this.state.favs);
-        }
-      );
-    } else {
-      this.setState(
-        state => {
-          const favs = state.favs.filter(id => id !== value);
-          return {
-            favs
-          };
-        },
-        () => {
-          this.saveInLocalStorage(this.state.favs);
-        }
-      );
-    }
+        return {
+          favs
+        };
+      },
+      () => this.saveInLocalStorage(this.state.favs)
+    );
```

Comment:
Looks fine, although maybe it would make more sense to decouple these functionallities:
* Remove fav
* Add new favourite
which would be more single responsibility principle friendly


Also IMO it would be better to explicitly pass ID and isFavourited to that handler
```
handleFavbutton = (id, isFav) => {
  if (isFav) {
    this.removeFromFavs(id);
  } else {
    this.addToFavs(id);
  }
}

And then anywhere else

onClick={() => handleFavButtonClicked(id, isFav)}
```

Just a thought, to make it clearer what this does.